crates/oxc_ast_lower/src/lib.rs

```diff
@@ -265,93 +273,105 @@ impl<'a> AstLower<'a> {
-            ast::Expression::ConditionalExpression(expr) => self.lower_conditional_expression(expr),
-            ast::Expression::FunctionExpression(expr) => self.lower_function_expression(expr),
-            ast::Expression::ImportExpression(expr) => self.lower_import_expression(expr),
-            ast::Expression::LogicalExpression(expr) => self.lower_logical_expression(expr),
-            ast::Expression::MemberExpression(expr) => self.lower_member_expression(expr),
-            ast::Expression::NewExpression(expr) => self.lower_new_expression(expr),
-            ast::Expression::ObjectExpression(expr) => self.lower_object_expression(expr),
-            ast::Expression::ParenthesizedExpression(expr) => {
-                self.lower_parenthesized_expression(expr)
-            }
-            ast::Expression::PrivateInExpression(expr) => self.lower_private_in_expression(expr),
-            ast::Expression::SequenceExpression(expr) => self.lower_sequence_expression(expr),
-            ast::Expression::TaggedTemplateExpression(expr) => {
-                self.lower_tagged_template_expression(expr)
-            }
-            ast::Expression::ThisExpression(expr) => self.lower_this_expression(expr),
-            ast::Expression::UnaryExpression(expr) => self.lower_unary_expression(expr),
-            ast::Expression::UpdateExpression(expr) => self.lower_update_expression(expr),
-            ast::Expression::YieldExpression(expr) => self.lower_yield_expression(expr),
-            ast::Expression::Super(expr) => self.lower_super(expr),
-            ast::Expression::JSXElement(elem) => {
-                // TODO: implement JSX
-                let ident = self.lower_identifier_reference(&ast::IdentifierReference {
-                    span: elem.span,
-                    name: "undefined".into(),
-                });
-                self.hir.identifier_reference_expression(ident)
-            }
-            ast::Expression::JSXFragment(elem) => {
-                // TODO: implement JSX
-                let ident = self.lower_identifier_reference(&ast::IdentifierReference {
-                    span: elem.span,
-                    name: "undefined".into(),
-                });
-                self.hir.identifier_reference_expression(ident)
-            }
-            // Syntax trimmed for the following expressions
-            ast::Expression::TSAsExpression(expr) => self.lower_expression(&expr.expression),
-            ast::Expression::TSSatisfiesExpression(expr) => self.lower_expression(&expr.expression),
-            ast::Expression::TSNonNullExpression(expr) => self.lower_expression(&expr.expression),
-            ast::Expression::TSTypeAssertion(expr) => self.lower_expression(&expr.expression),
-            ast::Expression::TSInstantiationExpression(expr) => {
-                self.lower_expression(&expr.expression)
-            }
-        }
+        ensure_sufficient_stack(|| {
```

Comment:
Would it be sufficient to only call into this for expressions that support nesting?

Reply:
I don't think so, rustc does this on the whole expr.

It is probably due to condition + binary + paren expresions can nest over each other during recursion.